### ISSUE TITLE
Recovering window title using TProcess sometimes gives incorrect value.

### ISSUE THREAD

**Issue by R030t1** [MerlijnWajer/Simba#408]
See: https://villavu.com/forum/showthread.php?t=117585&p=1386995#post1386995

**Comment by R030t1** [comment]
Per discussion in the thread the likely culprit is UTF-16 values inside of a Pascal string.

The concerns farther down in #338 are relevant to this bug report - the best way to handle Unicode support on Windows is to use UTF-8 internally and then convert it to UTF-16 when using OS facilities. If there is any way to put this in a unit that would be ideal, but this is something that needs to be done every place where Windows text API is used.

**Comment by R030t1** [comment]
The forum post contains an outline for a fix.

**Comment by R030t1** [comment]
Per [LCL documentation](http://wiki.lazarus.freepascal.org/Unicode_Support_in_Lazarus#WinAPI_function_calls_in_FPC_libs) TProcess uses the system codepage on Windows, meaning it is UTF-16. The result needs to be converted to UTF-8 for display with `WinCPToUTF8` (the `UnicodeString` functions may or may not work) or `TProcessUTF8` of the unit `utf8process` needs to be used. There may also be changes that need to be made to the project as a whole, see #338.

**Comment by WarPie** [comment]
Window title + TProcess, something is very off here... `TProcess` is class you use to start an "external" process, eg, to run "ls" in the terminal, or to execute a python script from Simba.

The villavu thread you are referring to talks about `TSysProc` *not* `TProcess`.  `TSysProc` is just a simple datastructure @JohnPeel made when he added the [`GetProcesses`](https://github.com/MerlijnWajer/Simba/blob/master/Units/MMLCore/os_windows.pas#L586) function to Simba.

https://github.com/MerlijnWajer/Simba/commit/58faccf18272fc0d7a5ead98d47fd1f93754298d

All that I see that is needed is the following steps: 
Change [`Title` field in TSysProc](https://github.com/MerlijnWajer/Simba/blob/master/Units/MMLCore/mufasatypes.pas#L162) to `WideString`. 
Change it also for the [lape export](https://github.com/MerlijnWajer/Simba/blob/master/Units/MMLAddon/LPInc/lpcompile.inc#L129)
And finally replace [this line](https://github.com/MerlijnWajer/Simba/blob/master/Units/MMLCore/os_windows.pas#L586) with something along the lines of:
`SetLength(ProcArr[I].Title, GetWindowTextW(Handle, PWideChar(ProcArr[I].Title), Length(ProcArr[I].Title)));`

**Comment by R030t1** [comment]
Yeah I kind of wonder what I was thinking. If nobody commits the change I will see if I can make one that doesn't use `WideString`.
